src/gui/scheduling/etagwatcher.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) by Hannah von Reth <hannah.vonreth@owncloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include "gui/scheduling/etagwatcher.h"
+
+#include "accountstate.h"
+#include "gui/folderman.h"
+#include "libsync/configfile.h"
+#include "libsync/graphapi/spacesmanager.h"
+#include "libsync/syncengine.h"
+
+
+using namespace std::chrono_literals;
+
+using namespace OCC;
+
+namespace {
+constexpr auto pollTimeoutC = 30s;
+}
+
+EtagWatcher::EtagWatcher(FolderMan *folderMan, QObject *parent)
+    : QObject(parent)
+    , _folderMan(folderMan)
+{
+    connect(folderMan, &FolderMan::folderListChanged, this, [this] {
+        decltype(_lastEtagJob) intersection;
+        for (auto *f : _folderMan->folders()) {
```

Comment:
Please add some explanatory comment what the purpose of this intersection is and why you have to manage it yourself here.

Reply:
Failed to parse the question.
We maintain a list of folders we can either add a folderAdded signal and subscribe it along with folderRemoved or we take the change event.